conans/client/graph/graph_binaries.py

```diff
@@ -295,6 +298,7 @@ def _process_node(self, node, pref, build_mode, update, remotes):
                     conanfile.output.info("Outdated package!")
                     node.binary = BINARY_BUILD
                     node.prev = None
+                    _call_validate_build(conanfile)
```

Comment:
I would try to call this earlier in the process, just once, as it doesn't have much effect besides setting an attribute, it should be possible.